e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/add-to-an-existing-payment.spec.js

```diff
@@ -67,24 +94,49 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can add fee records to existing paymen
   });
 
   it('should render the add to an existing payment page', () => {
-    cy.get('h1').invoke('text').should('contain', 'Add to an existing payment');
+    cy.get('h1').invoke('text').should('contain', 'Add reported fee to an existing payment');
   });
 
-  it('should render the selected fee record details', () => {
+  it('should render the selected fee records, render the available payments, and add fees to the payments', () => {
     pages.utilisationReportAddToAnExistingPaymentPage.selectedReportedFeesDetailsTable().should('contain', '11111111');
     pages.utilisationReportAddToAnExistingPaymentPage.selectedReportedFeesDetailsTable().should('contain', 'Exporter 1');
     pages.utilisationReportAddToAnExistingPaymentPage.selectedReportedFeesDetailsTable().should('contain', 'JPY 100');
     pages.utilisationReportAddToAnExistingPaymentPage.selectedReportedFeesDetailsTable().should('contain', 'GBP 50');
-
     pages.utilisationReportAddToAnExistingPaymentPage.selectedReportedFeesDetailsTable().contains('Total reported payments GBP 50').should('exist');
-  });
 
-  it('should render the available payments', () => {
     pages.utilisationReportAddToAnExistingPaymentPage.availablePaymentGroups().should('exist');
+    pages.utilisationReportAddToAnExistingPaymentPage.availablePaymentGroups().should('contain', 'GBP 450.00');
+    pages.utilisationReportAddToAnExistingPaymentPage.availablePaymentGroups().should('contain', 'Payment reference: REF00001');
+
+    pages.utilisationReportAddToAnExistingPaymentPage.paymentGroupRadioButton(`paymentIds-${PAYMENT_ID_ONE}`).click();
+
+    pages.utilisationReportAddToAnExistingPaymentPage.continueButton().click();
+
+    pages.utilisationReportPage.premiumPaymentsTab.getPaymentLink(PAYMENT_ID_ONE).should('contain', 'GBP 450.00');
+    pages.utilisationReportPage.premiumPaymentsTab.getPremiumPaymentsTableRow(FEE_RECORD_ID_ONE).should('contain', 'MATCH');
+  });
+
+  it('should automatically select the first payment group when there is only one group to choose from', () => {
+    cy.task(NODE_TASKS.REMOVE_ALL_UTILISATION_REPORTS_FROM_DB);
+    cy.task(NODE_TASKS.REMOVE_ALL_PAYMENTS_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_UTILISATION_REPORTS_INTO_DB, [report]);
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [feeRecordOne, feeRecordTwo]);
+
+    cy.reload();
```

Comment:
I'm torn about this, do we do this elsewhere? I think it seems a bit odd to essentially remove all the set up and then restart it and also I guess this is why we are inserting the report separately which also seems less ideal than inserting a report with the fee records attached... I don't know if we nest many contexts, but I wonder if we can have two contexts and then they have slightly differing `beforeEachs`?

Reply:
Refactored this into the two contexts with a helper function to perform the actions that are shared.